File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        char[] arrs = s.toCharArray();
        char[] arrt = t.toCharArray();
        Arrays.sort(arrs);
        Arrays.sort(arrt);
        if (arrs.length != arrt.length) return false;
        for (int i = 0; i < arrs.length; i++) {
            if (arrs[i] != arrt[i]) {
                return false;
            }
        }
        return true;
    }
}

File: 0242-valid-anagram/0242-valid-anagram.ts
function isAnagram(s: string, t: string): boolean {
    if (s.length !== t.length) {
        return false
    }
    const sObj = {};
    const tObj = {};


    for (let i = 0; i < s.length; i++) {
        sObj[s[i]] = (sObj[s[i]] || 0) + 1;
        tObj[t[i]] = (tObj[t[i]] || 0) + 1;
    }

    for (let key in sObj) {
        if (sObj[key] !== tObj[key]) {
            return false
        }
    }

    return true
};